openedx/frontend-lib-content-components
# feat: video editor fix to disappearing transcript

https://2u-internal.atlassian.net/browse/TNL-10300

Issue: Currently, when user updates the video id or video url field, the rest of the settings will get reset. This occurs onBlur, which means if the user updates one of these fields and immediately clicks Save, they will not realize the form has completely updated. Another point of confusion is that there is no notification that the settings are reset. 
Users filed this issue because importing youtube transcripts reset would set a video id, which would then reset the youtube video url.

Solution: Do not reset settings when video id or video url field is updated. This copies legacy behavior and means both id and url can be saved. User work is saved where possible

Testing:
1. Create a new video block with a video url that has transcripts (can use this one https://youtu.be/B-EFayAA5_0).
2. Import transcripts.
3. Studio page should not show "No playable video sources found" on that block.
4. Opening up the block again should show both a video id and a video url.

Minor Refactor Note: Biggest part of this was removing react-thunk as we are mostly not using it. We were using useSelector hook to pull form store anyways.
